shell/platform/windows/client_wrapper/include/flutter/flutter_view_controller.h

```diff
@@ -57,6 +57,14 @@ class FlutterViewController {
                                                   WPARAM wparam,
                                                   LPARAM lparam);
 
+  // Non-Flutter windows added manually to an application shall pass window
+  // messages from their WndProc functions to this method call in order to
+  // implement the application lifecycle for itself.
+  void HandleExternalWindowMessage(HWND hwnd,
+                                   UINT message,
+                                   WPARAM wparam,
+                                   LPARAM lparam);
```

Comment:
What's the reasoning for putting this on the `FlutterViewController`? Keep in mind that Flutter can be in a "headless" mode where there's no Flutter views, however, they may be external windows.

Reply:
It seemed reasonable to handle this in a similar manner to `HandleTopLevelWindowProc`. I am unsure of where else we would be able to expose this API call so it can be called from an app with an external window and in turn be able to access a/the `FlutterWindowsEngine` that holds the `WindowsLifecycleManager`. Are you aware of a better sounding alternative?

Reply:
I'd lean towards moving this to `FlutterEngine`.

Here's what a multi-view app will look like on Windows: https://github.com/goderbauer/mvp/pull/9/files#diff-f3c022dee233416d90c619b80526ec0cb3377657c88a45b2be0f34a7cf77a061R32

You'll be able to create a `FlutterEngine` and then pass it to the `FlutterViewController`'s constructor. For now, you can access the `FlutterEngine` using `FlutterViewController::engine`.

Reply:
That's reasonable. I'll make that change momentarily, thanks.

Reply:
FYI `FlutterWindowsEngine` isn't part of the Flutter Windows's public API. You'll need to expose this API using `//shell/platform/windows/client_wrapper/include/flutter/flutter_engine.h` and `//shell/platform/windows/public/flutter_windows.h`.

Reply:
@yaakovschectman It looks like this is still pending. Please let me know if I'm missing something!

Reply:
Please see if the new changes are in line with what you had in mind? Thanks.

Reply:
Yup, looks good!